core/src/main/java/com/datastrato/gravitino/listener/api/info/PartitionInfo.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datastrato.gravitino.listener.api.info;
+
+import com.datastrato.gravitino.annotation.DeveloperApi;
+import com.datastrato.gravitino.rel.partitions.Partition;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+/**
+ * Provides access to metadata about a Partition instance, designed for use by event listeners. This
+ * class encapsulates the essential attributes of a Partition, including its name and properties
+ * information.
+ */
+@DeveloperApi
+public final class PartitionInfo {
+  private final String name;
+  private final Map<String, String> properties;
+
+  /**
+   * Constructs PartitionInfo from an existing Partition object.
+   *
+   * @param partition The Partition instance to extract information from.
+   */
+  public PartitionInfo(Partition partition) {
```

Comment:
Partition could be `ListPartition` `RangePartition` and `IdentityPartition`,  seems we should create corresponding ParititionInfo to represent different partitions.

Reply:
This means that a separate PartitionInfo class will be created for each of the three data structures, e.g., ListPartitionInfo, RangePartitionInfo, and IdentityPartitionInfo?

Or is it that polymorphism will be implemented within the PartitionInfo class?

Reply:
I prefer to create seperate `ListPartitionInfo` , `RangePartitionInfo`, `IdentityPartitionInfo`  with the common base class `PartitionInfo`

Reply:
I have two questions

1. I'm unsure whether to **declare PartitionInfo as an interface or a class**, since other XXXInfo are declared as classes.
If declared as a class, the constructor in the child class would need to call the parent's constructor, which seems a bit ugly.

2. I don't know how to include RangePartitionInfo and others in PartitionEvent usage. PartitionDispatcher does not have any related calls, while Partitions provides a series of static builder methods for variations of partitions.
So this requires individual events and helper methods similar to Partitions?

Reply:
1. declare PartitionInfo as a class and I think it's normal to call parent's constructor in child‘s constructor.
2. we could use a simple static method to create diverse PartitionInfo, like:
```
public class PartitionInfo {
    public static PartitionInfo  create(Partition partition) {
       // not sure 
       if (partition instanceof IdentityPartition) {
           return new IdentityPartitionInfo(xxx)
       } else {xxxx}
    }
}
```

Reply:
OK, already add create method. One mroe question.

I'm not sure if Range/Identity/List PartitionInfo needs to be checked for all events. For example, AddPartitionEvent/DropPartitionEvent/etc.

Or does only GetPartitionEvent need to check the PartitionInfo type? Based on the SupportsPartitions.java file, only getPartition might return different Partition types.

Reply:
I think it should checked for all events with `PartitionInfo`